pkg/api/validate/admin.go

```diff
@@ -0,0 +1,21 @@
+package validate
+
+import (
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+// Validate validates a OpenShiftManagedCluster struct
+func (v *AdminAPIValidator) Validate(new, old *api.OpenShiftManagedCluster, externalOnly bool) (errs []error) {
+	if errs := validateContainerService(new, externalOnly); len(errs) > 0 {
```

Comment:
The first thing you should validate here is that oldCs is non-nil, otherwise you will return an error about the aad secret not being provided which is not available to admins at all. This is already implemented in the existing admin validation.

Reply:
One more reason to split as logic as becoming complicated...